### PR TITLE
Add schedule api

### PR DESCRIPTION
**Pull request by SubaMoorthy**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/machineLearningServices.json [hunk 1]
```diff
@@ -4584,6 +4640,209 @@
...
+    },
+    "ScheduleStatus": {
+      "type": "string",
+      "description": "The schedule status.",
+      "enum": [
+        "Enabled",
+        "Disabled"
+      ],
+      "x-ms-enum": {
+        "name": "status",
```
Comment (ArcturusZhang) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/machineLearningServices.json:4692]: This enum type conflicts with another, we need a new name for this

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

**Review by JeffreyRichter — APPROVED** [review]

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/machineLearningServices.json [hunk 2]
```diff
@@ -4584,6 +4640,209 @@
...
+        },
+        "expression": {
+          "type": "string",
+          "description": "The cron expression."
+        }
+      }
+    },
+    "ComputeSchedules": {
+      "type": "object",
+      "description": "The list of schedules to be applied on the Compute Instance",
```
Comment (ninghu) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/machineLearningServices.json:4836]: Change to "The list of schedules to be applied on the compute"?

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/machineLearningServices.json [hunk 3]
```diff
@@ -3155,6 +3208,9 @@
                   "description": "The last operation on ComputeInstance.",
                   "$ref": "#/definitions/ComputeInstanceLastOperation",
                   "readOnly": true
+                },
+                "computeSchedules": {
+                  "$ref": "#/definitions/ComputeSchedules"
```
Comment (ninghu) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/machineLearningServices.json:3213]: 1. add description
2. new schedules property should be readOnly

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/examples/Compute/createOrUpdate/ComputeInstanceWithSchedules.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          "computeSchedules": {
+            "computeStartStop": [
+              {
+                "status": "Enabled",
+                "triggerType": "Cron",
+                "cron": {
+                  "expression": "0 18 * * *",
+                  "startTime": "2021-04-23T1:30:00",
+                  "timeZone": "Pacific Standard Time"
+                }
```
Comment (ninghu) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/examples/Compute/createOrUpdate/ComputeInstanceWithSchedules.json:35]: missing action property

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/examples/Compute/updateSchedules.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            "timeZone": "Pacific Standard Time",
+            "schedule": {
+              "hours": [
+                18
+              ],
+              "minutes": [
+                30
+              ],
+              "weekDays": null
+            }
```
Comment (ninghu) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/examples/Compute/updateSchedules.json:26]: missing action property

**Review by ninghu — CHANGES_REQUESTED** [review]

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/machineLearningServices.json (hunk 3 above)
Comment (SubaMoorthy) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/machineLearningServices.json:3214]: Updated description. Not making it readonly as it is user input for create CI api.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/examples/Compute/createOrUpdate/ComputeInstanceWithSchedules.json (hunk 4 above)
Comment (ninghu) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/examples/Compute/createOrUpdate/ComputeInstanceWithSchedules.json:36]: Can we change to Stop in this example as it's a more common scenario?

**Review by ninghu — APPROVED** [review]

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/examples/Compute/updateSchedules.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    "parameters": {
+      "computeStartStop": [
+        {
+          "status": "Enabled",
+          "action": "Start",
+          "triggerType": "Recurrence",
+          "recurrence": {
+            "frequency": "Day",
+            "interval": 1,
+            "startTime": "2021-04-23T1:30:00",
```
Comment (ninghu) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/examples/Compute/updateSchedules.json:17]: I believe 2021-04-23T1:30:00 is not valid, it has to be 2021-04-23T01:30:00

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/examples/Compute/createOrUpdate/ComputeInstanceWithSchedules.json [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+          },
+          "schedules": {
+            "computeStartStop": [
+              {
+                "status": "Enabled",
+                "action": "Stop",
+                "triggerType": "Cron",
+                "cron": {
+                  "expression": "0 18 * * *",
+                  "startTime": "2021-04-23T1:30:00",
```
Comment (ninghu) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2021-03-01-preview/examples/Compute/createOrUpdate/ComputeInstanceWithSchedules.json:34]: Same here, not valid

**Review by ninghu — CHANGES_REQUESTED** [review]

**Review by ninghu — APPROVED** [review]

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**